servers/physics_2d/godot_body_2d.h

```diff
@@ -202,7 +202,8 @@ class GodotBody2D : public GodotCollisionObject2D {
 	_FORCE_INLINE_ uint64_t get_island_step() const { return island_step; }
 	_FORCE_INLINE_ void set_island_step(uint64_t p_step) { island_step = p_step; }
 
-	_FORCE_INLINE_ void add_constraint(GodotConstraint2D *p_constraint, int p_pos) { constraint_list.push_back({ p_constraint, p_pos }); }
+	// The newest constraint is added to the front to prevent incorrect populating when replacing the old constraint with the new one.
+	_FORCE_INLINE_ void add_constraint(GodotConstraint2D *p_constraint, int p_pos) { constraint_list.push_front({ p_constraint, p_pos }); }
```

Comment:
Since only `GodotBody2D` uses `List` (implemented by #44112), structures supporting FIFO/LIFO are not yet used in `GodotArea2D` and `GodotArea3D`/`GodotBody3D`/`GodotSoftBody3D`. 

So it is still random in the corresponding `CollisionObject`.